nginx-module.cmake

```diff
@@ -17,31 +17,46 @@ list(REMOVE_DUPLICATES NGINX_CONF_ARGS)
 
 ## Only for printing
 list(JOIN NGINX_CONF_ARGS " " NGINX_CONF_ARGS_LOG)
-
 message(STATUS "Nginx configuration args: ${NGINX_CONF_ARGS_LOG}")
 
 set(NGINX_MODULE_FILE ${NGINX_SRC_DIR}/objs/ngx_http_datadog_module_modules.c)
 set_source_files_properties(${NGINX_MODULE_FILE} PROPERTIES GENERATED TRUE)
 
-add_custom_command(
-  OUTPUT ${NGINX_MODULE_FILE}
-  COMMAND ./configure ${NGINX_CONF_ARGS}
-  WORKING_DIRECTORY ${NGINX_SRC_DIR}
-  COMMENT "Configuring nginx"
-)
+include(FetchContent)
 
-target_sources(nginx_module
-  PRIVATE
-    ${NGINX_MODULE_FILE}
-)
+set(NGINX_VERSION "" CACHE STRING "The nginx version")
```

Comment:
This could have a default value as well, in which case a warning would be sufficient. Even better if it were possible to download the latest stable version although unclear how that could be determined.

Reply:
Perhaps, but that is not the current behavior. You need to see NGINX_VERSION.